### DISCUSSION TITLE
GitLab sync - where it the file?

### DISCUSSION THREAD

**Discussion by Karol-Wi** [knqyf263/pet#291] [Q&A] [answered]
Hi!
First of all - I ❤️pet! It is sooooo useful!
I have a question related to GitLab sync. I've synchronized my snippets file with a GitLab. I have an ID of the snippet.
The problem is that I cannot see (or find) it in GitLab GUI. The snippets tab is empty, there is no new file in the repo.

Could anyone please help me with that?
Thanks!

**Comment by RamiAwar** [comment]
> Hi!
> First of all - I ❤️pet! It is sooooo useful!
> I have a question related to GitLab sync. I've synchronized my snippets file with a GitLab. I have an ID of the snippet.
> The problem is that I cannot see (or find) it in GitLab GUI. The snippets tab is empty, there is no new file in the repo.
> 
> Could anyone please help me with that?
> Thanks!

Hey @Karol-Wi, thanks for the ♥️🙂!

I've actually never tried the Gitlab sync myself, I usually use Github.

Let me try that out for you later today and see what's going on!

**Comment by RamiAwar** [comment]
Hey @Karol-Wi , so I tried it out and it worked fine for me.

Let's walk through your steps from the start again.

1. Delete your snippet on the Gitlab UI if you see it at all, if not forget about it.

2. Run `pet configure` and reset the Gitlab section to the following:

```
[GitLab]
  file_name = "pet-snippet.toml"
  access_token = "<YOUR ACCESS TOKEN>"
  url = "https://gitlab.com"
  id = ""
  visibility = "private"
  auto_sync = false
  skip_ssl = false
  ```
  and save the file and exit.
  
  3. Run `pet sync`. It should show something like this:
<img width="242" alt="image" src="https://github.com/knqyf263/pet/assets/8523191/19fa7c9c-de4e-4d7e-8a18-a62a048431e6">

Let me know what you get!

**Comment by Karol-Wi** [comment, marked as answer]
Hi @RamiAwar ,  👋 
thanks for the answer.

I've found the problem 🎉
I've created an access token **inside a project**. That is why Pet has created a snippet, but not attached it to the project. It is visible via API, but not in GUI. Which is a bit odd. 

Once I've created an access token **outside of projects** - it works like a charm 👍 
The snippet is visible here: https://gitlab.com/dashboard/snippets

Thanks for your help! 💪
